frontend/src/components/profile/my-tickets.tsx

```diff
@@ -0,0 +1,264 @@
+import { useCurrentLanguage } from "~/locale/context";
+import {
+  AttendeeTicket,
+  useUpdateTicketMutation,
+  MyProfileDocument,
+} from "~/types";
+
+import { ProductState } from "../tickets-page/types";
+
+type Props = {
+  tickets: AttendeeTicket[];
+};
+
+export const MyTickets = ({ tickets = [] }: Props) => {
+  const code = process.env.conferenceCode;
+
+  const language = useCurrentLanguage();
+  const { user } = useCurrentUser({ skip: false });
+  const ticketHeader = useTranslatedMessage("profile.ticketFor");
+  const nameHeader = useTranslatedMessage("orderReview.attendeeName");
+  const emailHeader = useTranslatedMessage("orderReview.attendeeEmail");
+  const [currentTicketId, setCurrentTicketId] = useState(null);
+  const [error, setError] = useState(null);
+
+  const headers = [ticketHeader, nameHeader, emailHeader, ""];
+
+  const [selectedProducts, setSelectedProducts] = useState(
+    tickets.reduce((accumulator, ticket) => {
+      // if accumulator[ticket.id] is undefined assign empty list
+      (accumulator[ticket.id] = accumulator[ticket.id] || []).push({
+        id: ticket.id,
+        attendeeName: ticket.name,
+        attendeeEmail: ticket.email,
+        answers: Object.fromEntries(
+          ticket.item.questions.map((question) => [
+            question.id,
+            question.options.length > 0
+              ? question.answer.options[0]
+              : question.answer.answer,
+          ]),
+        ),
+      } as ProductState);
+      return accumulator;
+    }, {}),
+  );
+
+  const [
+    updateTicket,
+    { data: updatedData, loading: updatingTicket, error: updatedError },
+  ] = useUpdateTicketMutation({
+    onCompleted(result) {
+      if (
+        ["AttendeeTicket", "TicketReassigned"].indexOf(
+          result.updateAttendeeTicket.__typename,
+        ) > -1
+      ) {
+        setCurrentTicketId(null);
+      }
+    },
+    update(cache, { data }) {
+      if (data.updateAttendeeTicket.__typename === "TicketReassigned") {
+        setCurrentTicketId(null);
+        const { me } = cache.readQuery({
+          query: MyProfileDocument,
+          variables: {
+            conference: process.env.conferenceCode,
+            language: language,
+          },
+        });
+        cache.writeQuery({
+          query: MyProfileDocument,
+          data: {
+            me: {
+              ...me,
+              tickets: me.tickets.filter(
+                (ticket) => ticket.id !== data.updateAttendeeTicket.id,
+              ),
+            },
+          },
+          variables: {
+            conference: process.env.conferenceCode,
+            language: language,
+          },
+        });
+      }
+    },
+    onError(err) {
+      console.error(err);
+      setError(err.message);
+    },
+  });
+
+  const updateTicketCallback = useCallback(
+    (id: string) => {
+      return () => {
+        const answers = tickets
+          .filter((item) => item.id == id)[0]
```

Comment:
all of those hard coded `[0]` around this code make me feel a bit uneasy 😓

Reply:
yeah, in order to make it work with `QuestionsSections` I have these weird structures... Also, because I render only one ticket at a time, not all the tickets like on the `/tickets` page...

Reply:
On the `/tickets` page we group by the `TicketItem.id` so we can have many items for one id, here I grouped by the `AttendeeTicket.id` that is unique, so there will be only one item.